### PR TITLE
Enumerables: Release/0.1.0

### PR DESCRIPTION
**Pull request by Kasuhira**
Hello, reviewer, how are you?
I'm releasing the first version of the Enumerables algoritms, the focus is to make it usable and learn about git flow, I'm having dificulties on finding more bugs so I ask for guidance.
When ready, I shall merge, happy review !

### CODE REVIEW

File: index.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+  def my_count
+    counter = 0
+    if block_given?
+      my_each { |x| counter += 1 if yield(x) }
+      counter
+    else
+      my_each { counter += 1 }
+    end
+  end
```
Comment (whiz25) [inline index.rb]: - [ ] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns the number of items in enum through enumeration
- [ ] counts the number of items in enum that are equal to item if an argument is given

File: index.rb [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        end
+        ind
+      else
+        "undefined method for #{ind}:#{ind.class}"
+      end
+    else
+      'no block given (LocalJumpError)'
+    end
+  end
+end
```
Comment (whiz25) [inline index.rb]: - [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

File: enumerable_methods.rb [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      array = []
+      my_each { |x| array.push(proc.call(x)) }
+      array
+    else
+      to_enum(:my_map)
+    end
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def my_inject(ind = nil, symb = nil)
```
Comment (Forison) [inline enumerable_methods.rb]: * [ ] This example is failing, kindly address them
```
longest = %w{ cat sheep bear }.my_inject do |memo, word|
   memo.length > word.length ? memo : word
end
p longest 
longest = [ 'cat', 'sheep', 'bear' ].my_inject do |memo, word|
   memo.length > word.length ? memo : word
end
p longest 

```
Check [Ruby doc](https://ruby-doc.org/core-2.7.0/Enumerable.html)

File: enumerable_methods.rb [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    counter = 0
+    if block_given?
+      my_each { |x| counter += 1 if yield(x) }
+    else
+      my_each do |x|
+        counter += 1 if num == x
+      end
+    end
+    counter
+  end
```
Comment (stratospherique) [inline enumerable_methods.rb]: * [ ] my_count returns the number of items in enum through enumeration
  ```ruby
    array = [1,2,5,4,7]
    p array.my_count == array.count # returns this error
      #
      # ArgumentError:
       wrong number of arguments (given 0, expected 1)
     # ./enumerable_methods.rb:93:in `my_count'
  ```
* [ ] my_count counts the number of elements yielding a true value if a block is given
```ruby
array = [1,2,8,9,7,5,6,3,11]
block = proc { |num| num < 5 }
array.my_count(&block) == array.count(&block) # return this error
   # ArgumentError:
       wrong number of arguments (given 0, expected 1)
     # ./enumerable_methods.rb:93:in `my_count'
```

File: enumerable_methods.rb [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        end
+        ind
+      else
+        "undefined method for #{ind}:#{ind.class}"
+      end
+    else
+      'no block given (LocalJumpError)'
+    end
+  end
+end
```
Comment (stratospherique) [inline enumerable_methods.rb]: * [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
  ```ruby
    array = [4,7,8,89,9,6,5,12,4,2,5]
    operation = proc { |sum, n| sum + n  }
    p array.my_inject(&operation) == array.inject(&operation) # false
  ```
* [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```ruby
    range = Range.new(5, 50)
    p range.my_inject(2, :*) # returns this error
    # NameError:
       undefined local variable or method `length' for 5..50:Range
     # ./enumerable_methods.rb:6:in `my_each'
     # ./enumerable_methods.rb:149:in `my_inject'
  ```

**Pull request closed without merging**